### PR TITLE
Ditch render-modifiers

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
WIP - exploration for now

Begins #88

### CODE REVIEW

File: addon/utils/function-modifier.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  element?: HTMLElement;
+  destructor?: void | (() => void);
+}
+
+/**
+ *  This is copied from
+ *  https://github.com/pzuraq/ember-could-get-used-to-this/blob/master/addon/-private/modifiers.js
+ *
+ * Because:
+ *  - we don't want to *force* folks into use ember-could-get-used-to-this
```
Comment (GavinJoyce) [inline addon/utils/function-modifier.ts:34]: No against including this here, just wondering why we don't want to depend on `ember-could-get-used-to-this`?

Comment (NullVoxPopuli) [reply addon/utils/function-modifier.ts:34]: we absolutely could! what other things might we want to use `ember-could-get-used-to-this` for?
(I think we need to move ember-could-get-used-to-this in to adopted-addons, or specifically extract out the function-modifier bit to a new addon and have that included everywhere -- maybe call it `ember-default-modifier-manager-polyfill` or something related to the matching RFC 🤷)

**Comment by NullVoxPopuli** [comment]
superseded by https://github.com/GavinJoyce/ember-headlessui/pull/116

**Comment by NullVoxPopuli** [comment]
superseded by: https://github.com/GavinJoyce/ember-headlessui/pull/116

**Pull request closed without merging**